magento/magento2
# reference directive restored for _extends.less. extends replaced to mixins in _extend.less

### Description (*)

When I installed a clear M2, I saw that the `reference` directive removed for the import of **_extends.less** file. I've found [this PR](https://github.com/magento/magento2/pull/31793) and I explained why removing `reference` is a bad idea:
https://github.com/magento/magento2/pull/31793#issuecomment-1038102831

I've investigated later how to restore `reference` but repair the issue of Grunt in [the PR](https://github.com/magento/magento2/pull/31793). 

And I've found a solution described here:
https://github.com/magento/magento2/pull/31793#issuecomment-1038905078

@ihor-sviziev asked me to create a PR with the fix, so I made it.

### Related Pull Requests
https://github.com/magento/magento2/pull/31793

### Manual testing scenarios (*)

1. Create a new theme which extends Magento Blank theme.
2. Compile styles via default deploying (setup:static-content:deploy) and via Grunt.
3. Look at block titles of My Account (they should look the same after all build processes).
4. Find matches of `abs` selectors in `styles-m.css` and `styles-l.css` files. Quantity of them and the size of `styles-m.css` should be less in comparison with the main branch (2.4-develop).

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#36914: reference directive restored for _extends.less. extends replaced to mixins in _extend.less
